packages/frontend/component/src/ui/scrollbar/use-has-scroll-top.tsx

```diff
@@ -25,7 +25,7 @@ export function useHasScrollTop(ref: RefObject<HTMLElement> | null) {
     return () => {
       container.removeEventListener('scroll', updateScrollTop);
     };
-  }, [ref]);
+  }); // avoid using ref.current as a dependency to prevent it from being set too late.
```

Comment:
Feeling will be a infinite loop re-render

Reply:
changed to another solution

Reply:
deps won't affect the time of effect get called. there must be something wrong if it only work when no deps set